src/rez/wheel.py

```diff
@@ -0,0 +1,527 @@
+    assert isinstance(names, (tuple, list)), "%s was not list or tuple" % names
+
+    tempdir = tempfile.mkdtemp(suffix="-rez", prefix="pip-")
+
+    distributions = download(
+        names,
+        tempdir=tempdir,
+        no_deps=no_deps,
+    )
+
+    packagesdir = prefix or (
+        config.release_packages_path if release
+        else config.local_packages_path
+    )
+
+    new, existing = list(), list()
+    for dist in distributions:
+        package = convert(dist, variants=variants)
+
+        item = {
+            "distribution": dist,
+            "package": package,
+        }
+
+        if exists(package, packagesdir):
+            existing.append(item)
+        else:
+            new.append(item)
+
+    if not new:
+        return []
+
+    for item in new:
+        deploy(item["distribution"], item["package"], path=packagesdir)
+
+    shutil.rmtree(tempdir)
+    return [item["package"] for item in new]
+
+
+def download(names, tempdir=None, no_deps=False):
+    """Gather pip packages in `tempdir`
+
+    Arguments:
+        names (list): Names of packages to install, in pip-format,
+            e.g. ["six==1"]
+        tempdir (str, optional): Absolute path to where pip packages go until
+            they've been installed as Rez packages, defaults to the cwd
+        no_deps (bool, optional): Equivalent to pip --no-deps, default to False
+
+    Returns:
+        distributions (list): Downloaded distlib.database.InstalledDistribution
+
+    Raises:
+        OSError: On anything gone wrong with subprocess and pip
+
+    """
+
+    assert isinstance(names, (list, tuple)), (
+        "%s was not a tuple or list" % names
+    )
+    assert all(isinstance(name, _basestring) for name in names), (
+        "%s contained non-string" % names
+    )
+
+    tempdir = tempdir or os.getcwd()
+
+    # Build pip commandline
+    cmd = [
+        "python", "-m", "pip", "install",
+        "--target", tempdir,
+
+        # Only ever consider wheels, anything else is ancient
+        "--use-pep517",
+
+        # Handle case where the Python distribution used alongside
+        # pip already has a package installed in its `site-packages/` dir.
+        "--ignore-installed",
+
+        # rez pip users don't have to see this
+        "--disable-pip-version-check",
+
+        # Prevent user-settings from interfering with install
+        "--isolated",
+    ]
+
+    if no_deps:
+        # Delegate the installation of dependencies to the user
+        # This is important, as each dependency may have different
+        # requirements of its own, and variants to go with it.
+        cmd.append("--no-deps")
+
+    cmd += names
+
+    popen = subprocess.Popen(cmd,
+                             stdout=subprocess.PIPE,
+                             stderr=subprocess.STDOUT,
+                             universal_newlines=True)
+
+    output = []
+    for line in iter(popen.stdout.readline, ""):
+
+        if line.startswith("DEPRECATION"):
```

Comment:
In the context of `rez wheel`, that's a `pip` implementation detail. But more importantly, there isn't anything the `rez wheel` user can do about it. :( At the moment, all it does is distract from the actual problem the package author or user needs to see, such as whether a pip version exists or whether there isn't any internet.

Reply:
Still, we should avoid filtering out deprecation warnings. Because one could be not related to python 2.7 end of support.

>In the context of rez wheel, that's a pip implementation detail.

I don't really agree. It's far from an implementation detail for a developer. A developer should be able to live with deprecation warnings and other warnings in general. There are there for a reason. Yes they are annoying, but they are important. And I also expect someone using `rez-pip` to know how to use pip and so know what kind of message he is suppose to see (since he will get the warnings anyway without rez).

Also, I think the PyPA org was planning to let PyPI return warnings so that pip, pipenv, and others would receive some warnings to pass to the user. That way, the python community is able to pass their messages without having people update their tools. These messages are important.